### PR TITLE
Warn if throw-keyids is set

### LINKED ISSUE #1756: RFE: detect when encryption recipients aren't being set
### Summary

As reported in https://github.com/gopasspw/gopass/issues/1083, it is possible for settings in `gpg.conf` to result encryption recipients not being recorded for gopass entries.  While this doesn't stop gopass from working with gpg, it does break other use cases.  Furthermore, gopass doesn't currently detect that attempts to fix/repair the password entries are failing.

### Steps To Reproduce
* Set `throw-keyids` in your `gpg.conf`. 

```
~/.gnupg/gpg.conf
...
# Disable putting recipient key IDs into messages
throw-keyids
```
* Create a new password entry
* run `gopass fsck --decrypt`
* run `gopass fsck --decrypt`

### Expected behavior
For gopass to detect that attempts to include the encryption recipients are failing.  When detected, possibly mention the problematic `gpg.conf` setting.

### Environment

- OS: Arch Linux
- OS version: rolling
- gopass Version: 1.11.0
- Installation method: Arch repository

### Additional context
None

### PR DESCRIPTION
**Pull request by dominikschulz**
Fixes #1756

Signed-off-by: Dominik Schulz <dominik.schulz@gauner.org>

### CODE REVIEW

**Comment by AnomalRoil** [comment]
IMO this should be documented somewhere, because people that do not include key ids are typically doing it on purpose to try and stay anonymous and we don't want to undermine their effort if they think it is important for them, do we?

**Comment by dominikschulz** [comment]
Good point, yes.
But this only affects gopass secrets.
Yes, people might want to apply this to secrets as well, but it just doesn't work well with our expectations.

**Comment by AnomalRoil** [comment]
Well I disagree, this works well with Gopass since GPG will simply try ever secret key in the key ring during a decryption attempt if this option was used upon encryption.

So Gopass itself is not hindered in anyway by the throw-id setting.

**Comment by dominikschulz** [comment]
The point is that some operations won't work as expected.

With age it's similar and there is not option to disable that.

So maybe we should only warn instead?
Or just ignore this setting and let the user figure it out?

**Comment by dominikschulz** [comment]
Updated the PR to only warn about throw-keyids, not override it.

File: internal/backend/crypto/gpg/cli/gpg.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ type GPG struct {
 	pubKeys   gpg.KeyList
 	privKeys  gpg.KeyList
 	listCache *lru.TwoQueueCache
+	throwKids bool
```
Comment (AnomalRoil) [inline internal/backend/crypto/gpg/cli/gpg.go:36]: Poor kids 😆

**Review by AnomalRoil — APPROVED** [review]

File: internal/backend/crypto/gpg/cli/gpg.go (hunk 1 above)
Comment (dominikschulz) [reply internal/backend/crypto/gpg/cli/gpg.go:36]: No kids were harmed in making this PR ;)

**Pull request merged**